### PR TITLE
Update design to be responsive

### PR DESCRIPTION
**Pull request by MikeRogers0**
Issue: https://github.com/lrug/lrug.org/issues/17

# What this changes

This is a small step towards making the design much nicer. 

It adds the following:
1. CSS Broken up into multiple files. It should make refactoring much easier in the future.
2. Uses CSS Variables - In the future we could easily add support for a dark mode with them.
3. Uses CSS Grid to add a responsive design. On mobiles it becomes single column.

What it doesn't do:
1. Change the design much - I didn't want to over do it, we can tackle little bits at a time.
2. Add Webpacker / PostCSS - I really want to use them (especially for assets like normalize.css), but for now I'm keeping it simple :)

# Screenshots

https://www.browserstack.com/screenshots/901b80174452083d34ad3a0095b424d7dd83209c - I used BrowserStack to view in some older browsers. It looks ok.

### CODE REVIEW

File: source/layouts/base.erb [hunk 1]
```diff
@@ -19,9 +20,7 @@
     <link rel="shortcut icon" href="https://assets.lrug.org/favicon.ico" />
     <link rel="icon" href="https://assets.lrug.org/favicon.ico" />
 
-    <%= stylesheet_link_tag "normalize", "all" %>
-    <%= javascript_include_tag  "all" %>
```
Comment (MikeRogers0) [inline source/layouts/base.erb:23]: The JS was totally empty - So I removed it :)

File: source/layouts/nights.erb [hunk 2]
```diff
@@ -7,7 +7,7 @@
     <meta content="IE=edge,chrome=1" http-equiv="X-UA-Compatible">
     <title>LRUG Nights : <%= page_title || "solvin' crimes, drinkin' beers" %></title>
     <link href="/rss/nights/" rel="alternate" title="RSS" type="application/rss+xml" />
-    <%= stylesheet_link_tag "normalize", "nights" %>
+    <%= stylesheet_link_tag "nights" %>
```
Comment (MikeRogers0) [inline source/layouts/nights.erb:10]: I moved the CSS over to SCSS and used an `@import` to include the `normalize` file. It reduces the request count by 1 :)

File: source/stylesheets/all.css [hunk 3]
```diff
@@ -1,221 +1,6 @@
-@charset "utf-8";
```
Comment (MikeRogers0) [inline source/stylesheets/all.css:1]: I left this file in here while I move things over.

**Comment by MikeRogers0** [comment]
@h-lame Is this along the right lines?

**Review by h-lame — COMMENTED** [review]
Thanks for this @MikeRogers0.  No real objections to SCSS over CSS, so by all means, continue down that path to simplify the source if you can.  

As for the design, it _does_ look better on narrow screens for sure, moving the sidebar out of the way really helps here.  I'm not sure the wider version feels "done" to me though.  It feels too wide in general, with too much space between sidebar and the main content.  Weirdly, even with the gap both the sidebar and main content feel too wide, the main content in particular.  The existing site design is almost certainly too tight, but there's probably a middle ground - perhaps just reducing the number of columns each part spans, or reducing the column widths in general?

As I scrutinise the design and compare it to what we have, there's definitely stuff I'd like to tidy up, not least being what's in the sidebar to give a better idea of what is current and useful to know about LRUG, and what is now a relic of a time gone by that we should still link to somewhere, but maybe not in the main sidebar.  I don't know if those kinds of thoughts are useful now or not as I suspect the main benefit here will be more modern tooling, and a site that looks better on smaller narrow screens.  We can tackle the information architecture later I suspect.

File: source/stylesheets/base/_layout.scss [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ *   #main {}
+ *     #content-wrapper {}
+ *       #content {}
+ *     #sidebar-wrapper {}
+ *       #sidebar {}
+ *   #footer {}
+ */
+
+*, :before, :after {
+  box-sizing: border-box;
```
[the remark this answers is not in the document]
Comment (MikeRogers0) [reply source/stylesheets/base/_layout.scss:13]: So that's present in both [Bootstrap](https://github.com/twbs/bootstrap/blob/ebad168f4296e83e36ee889667ba8907c1a93677/scss/_reboot.scss#L16-L20) & [TailwindCSS](https://github.com/tailwindlabs/tailwindcss/blob/2b0b306a77cc0f34cd25054a0d661a18278786bc/src/plugins/css/preflight.css#L99), I added it because I was a little thrown off by it not being present. 

[Bootstrap has documented their rational](https://getbootstrap.com/docs/4.1/getting-started/introduction/#box-sizing), which I do agree with. I think perhaps I need to setup a more explicit for this kind of change?

File: source/layouts/base.erb [hunk 5]
```diff
@@ -19,9 +20,7 @@
...
-
+    <%= stylesheet_link_tag "site" %>
```
[the remark this answers is not in the document]
Comment (MikeRogers0) [reply source/layouts/base.erb:23]: Awesome I can switch that back :) I think I just renamed it to make breaking out the new styles side by side & didn't know how readme.lrug.org pulled styled in.

That said! In the future I should probably aim to spend some time updating the readme to be independent - It could limit our options if we do want to embrace cache breakers & webpack.

File: source/stylesheets/base/_variables.scss [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+  --space-16: 4rem;
+
+  // Site 
+  --page-min-width: 320px;
+  --page-max-width: 1280px;
+
+  // Breakpoints - Annoyingly you can't use them within media queries, but here they are anyway
+  // --screen-md: 1024px;
+  // --screen-lg: 1280px;
+}
```
[the remark this answers is not in the document]
Comment (MikeRogers0) [reply source/stylesheets/base/_variables.scss:18]: Do you know the current browser trends for the LRUG site?

My reasoning behind wanting to use CSS Variables is:
1. [https://hey.com](https://hey.com) uses them.
2. They make creating a "Dark Theme" a lot easier in the future.
3. PostCSS has a nice plugin for [adding fallbacks for CSS Variables](https://github.com/postcss/postcss-custom-properties).

However, I'm happy to use SCSS variables & revisit CSS Variable usage in the future :)

File: source/stylesheets/base/_layout.scss [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  padding-right: var(--space-8);
+}
+
+#header {
+  //margin-bottom: var(--space-6);
+}
+
+#main {
+  overflow: hidden;
+  display: grid;
```
[the remark this answers is not in the document]
Comment (MikeRogers0) [reply source/stylesheets/base/_layout.scss:31]: I'll get some browser shots for us, I think it'll fall back to a single column layout, which should be reasonable :)

File: source/stylesheets/base/_variables.scss (hunk 6 above)
[the remark this answers is not in the document]
Comment (MikeRogers0) [reply source/stylesheets/base/_variables.scss]: I switched over to SCSS variables. We can revisit at CSS Variables in the future :)

File: source/stylesheets/base/_variables.scss [hunk 8]
```diff
@@ -1,18 +1,17 @@
...
-  --page-max-width: 1280px;
+// Site 
+$page-min-width: 320px;
+$page-max-width: 1280px;
 
-  // Breakpoints - Annoyingly you can't use them within media queries, but here they are anyway
-  // --screen-md: 1024px;
-  // --screen-lg: 1280px;
-}
+// Breakpoints - Annoyingly you can't use them within media queries, but here they are anyway
```
Comment (h-lame) [inline source/stylesheets/base/_variables.scss:15]: Does this comment still apply, now we're using SCSS variables?  I assume not because they're interpolated at compile time so we _should_ be able to use them in media queries, right?

File: source/stylesheets/old-all.css [hunk 9]
```diff
@@ -1,29 +0,0 @@
-#info {
-  text-align: right;
-}
```
Comment (h-lame) [inline source/stylesheets/old-all.css]: We didn't pull this over, but it looks like it might no longer be used.  Or it's a typo for `.info` which is used occasionally.  On reflection though, I'm not sure I'd want the things with `info` class to be right aligned.

tl;dr - dropping this style seems fine.

File: source/stylesheets/base/_reboot.scss [hunk 10]
```diff
@@ -1,3 +1,21 @@
 *, :before, :after {
   box-sizing: border-box;
 }
+
+@media print {
```
Comment (h-lame) [inline source/stylesheets/base/_reboot.scss:5]: If we're going all in on organisational efforts for the CSS, should these (and `#sidebar-wrapper @media print { ... }`) live in a `_print` file?

**Comment by h-lame** [comment]
> I reduced the maximum width, it's still a little wider then the current site but not by much :)

I still think the main content is too wide, but this could be a me thing - I'll see if the other organisers have any reckons.  I more confident in my feeling that the gap between the sidebar and content is too much.  How about reducing `grid-gap` on `#main` to `2rem` or even `1.5rem` ?

Thanks for taking the time to work on this, really appreciate it!

**Comment by h-lame** [comment]
> I'd be open to hearing what others want out of the site the next meetup :) 

I'd be surprised if we got any traction at the meeting itself, but we certainly could try during the "announcements" section between the talks.  It might be worth putting together some kind of survey?

> I'm still very happy to spend a few days here & there to help make it better.

TBH, as someone willing to put in the effort, it might be worth you writing down what _you_ want out of the site, and what other things you think we could change to make it better (including, I think, what "better" means here).

File: source/stylesheets/base/_variables.scss (hunk 8 above)
Comment (MikeRogers0) [reply source/stylesheets/base/_variables.scss:15]: Oh you're right! Updated!

File: source/stylesheets/old-all.css (hunk 9 above)
Comment (MikeRogers0) [reply source/stylesheets/old-all.css]: Awesome! I wasn't sure what to do with this one, I landed on just removing it as it wasn't being used :)

File: source/stylesheets/base/_reboot.scss (hunk 10 above)
Comment (MikeRogers0) [reply source/stylesheets/base/_reboot.scss]: I moved these closer to their parent definitions :) I'm not so keen on having a `_print` file, historically I've ended up forgetting to update them.

**Comment by MikeRogers0** [comment]
Awesome, I've actioned your feedback & I think we're very much there :)

> I more confident in my feeling that the gap between the sidebar and content is too much. How about reducing grid-gap on #main to 2rem or even 1.5rem ?

I updated it to be `1.5rem` - It looks a-ok, so I'm happy :)

 > I'd be surprised if we got any traction at the meeting itself, but we certainly could try during the "announcements" section between the talks. It might be worth putting together some kind of survey?

I might be a little pushed to put a survey together for Monday's LRUG. I'll try to put together something for later in the month we could possibly tweet out?

> TBH, as someone willing to put in the effort, it might be worth you writing down what you want out of the site, and what other things you think we could change to make it better (including, I think, what "better" means here).

Cool! I'll write down some ideas & see what sticks :)

**Comment by lazyatom** [comment]
It's great to have the site work better in smaller viewports, thanks for that  👍🎉

It would be good still show the logo somehow on mobile; for better or worse, it's a key part of the site identity now, and without it we just have text and red.

I think we could also reduce the font and padding size on smaller viewports a little -- perhaps just by making the `.first.entry` font size boost go away, and reducing the left/right padding from 2em to 1em.

**Comment by MikeRogers0** [comment]
> It would be good still show the logo somehow on mobile; for better or worse, it's a key part of the site identity now, and without it we just have text and red.

@lazyatom Where would you put the logo when on the mobile viewport? Maybe on mobile we put it above the header or something?

> I think we could also reduce the font and padding size on smaller viewports a little -- perhaps just by making the `.first.entry` font size boost go away, and reducing the left/right padding from 2em to 1em.

@lazyatom Yeah, I think that'll work nicely :) Do we want to remove the font size boost on all viewports, or just mobile?

**Comment by lazyatom** [comment]
> Where would you put the logo when on the mobile viewport? Maybe on mobile we put it above the header or something?

Maybe a small version just to the left of the main "LRUG" title? 

> Do we want to remove the font size boost on all viewports, or just mobile (Or have it boost for all entries so it feels more consistent)?

I would limit the change to only removing the boost on mobile; I'm not super attached to the boost in general, but I think there's value in taking small steps.

**Comment by MikeRogers0** [comment]
@lazyatom - I've pushed up the changes you requested. I did centre the logo on mobile views, when it was left/right it made the LRUG text look off centre.

image

File: source/_header.erb [hunk 11]
```diff
@@ -1,9 +1,12 @@
 <div id="header">
   <div id="site-title">
-<span class="line"><%= link_to 'LRUG', '/' %></span> : <span class="line">london</span> <span class="line ruby"><%= link_to 'ruby', 'http://www.ruby-lang.org/en/' %></span> <span class="line">user</span> <span class="line">group</span>
```
Comment (MikeRogers0) [inline source/_header.erb:3]: I wasn't sure why `class="line"` was used here. In the CSS they we're `display: inline`. I think it's a relic of a previous design maybe.

File: source/stylesheets/base/_layout.scss (hunk 7 above)
[the remark this answers is not in the document]
Comment (MikeRogers0) [reply source/stylesheets/base/_layout.scss:28]: https://www.browserstack.com/screenshots/901b80174452083d34ad3a0095b424d7dd83209c - IE Browser shots. It looks ok :)

[the remark this answers is not in the document]
Comment (h-lame) [reply source/stylesheets/base/_layout.scss:28]: Looks ok, thanks for checking!

File: source/stylesheets/base/_reboot.scss (hunk 10 above)
Comment (h-lame) [reply source/stylesheets/base/_reboot.scss]: Fair enough - keeping them with the parent seems like a good compromise.

File: source/_header.erb (hunk 11 above)
Comment (h-lame) [reply source/_header.erb:3]: Oh, that _is_ weird.  I suspect you're right and they're from a previous design as they're not necessary at all with the current design.

**Review by h-lame — APPROVED** [review]
Looks good - thanks for taking this on and moving us into the modern era of CSS techniques and organisation!
